crates/server/src/authority/catalog.rs

```diff
@@ -92,6 +95,29 @@ impl RequestHandler for Catalog {
                 };
             }
 
+            // RFC 5001 "DNS Name Server Identifier (NSID) Option" handling.
+            match (req_edns.option(EdnsCode::NSID), &self.nsid_payload) {
+                // "A name server MUST NOT send an NSID option back to a resolver which
+                // did not request it."
+                (None, ..) => {}
```

Comment:
Nit: IMO `_` is more idiomatic than `..` for this.

Reply:
Fixed.